### ISSUE TITLE
Search Revamp

### ISSUE THREAD

**Issue by brandonhaslegs** [drips-network/app#715]
If I search for ethers.js in the search bar, I should be able to find it. It's a claimed project, so it exists in the Drips system, so I should be able to find it.

<img width="528" alt="image" src="https://github.com/drips-network/app/assets/4406983/cce3f2f3-cb46-419a-aa99-fa4e94b48fb8">

<img width="447" alt="image" src="https://github.com/drips-network/app/assets/4406983/beede515-9158-4327-b333-a3395ff390c1">

**Comment by evvvritt** [comment]
I think we're also missing being able to search Drip Lists (by title)

**Comment by brandonhaslegs** [comment]
Yep should be a larger effort to improve search.

**Comment by efstajas** [comment]
Re searching GitHub repos by name:

This one unfortunately probably would be quite tricky, unless i'm missing some other approach. If the user just enters the repo name, we need to somehow find out the name of the owner's repo on GH in order to be able to locate it. So we basically would have to integrate with GitHub's [search endpoint](https://docs.github.com/en/rest/search/search?apiVersion=2022-11-28), but it's rate-limited to only 10 requests / minute if unauthenticated. If authenticated (meaning the user would have to log in to Drips with GitHub), the limit is still only 30 requests / minute.

I think 10 searches / minute is way too low for this to be worth implementing, unfortunately. 30 / minute is better, but then we'd need to make people log in to their GitHub account on Drips… no idea what that would look like.

A completely different approach could be searching projects through our own custom Drips GQL API soon, but… it only knows about already-claimed projects. So, the search bar would require you to enter a full URL for unclaimed projects, but could find e.g. "ethers.js" fine, since it's already claimed.

Of course regardless of all of this, we could make it so that you "only" need to enter `username/repo-name" instead of the full GH URL.

Curious to hear your thoughts!

**Comment by efstajas** [comment]
> search Drip Lists (by title)

we can do once our new API is up 👍

**Comment by efstajas** [comment]
Yeah, this we can do for sure 👍 

@jtourkos Just FYI — nothing we need to add now, but a "project name includes" string-filter for the projects query on the new API would probably be initially good enough for searching projects by name.

**Comment by brandonhaslegs** [comment]
Requirements:
- [ ] max 600 height, internal scroll
- [ ] arrow keys to select
- [ ] when not focused on current window, search disables. should be fixed while building this feature

<img width="736" alt="image" src="https://github.com/user-attachments/assets/53956b95-f2e3-40eb-ad81-e9072e88cb97">

or might be better to just show the USD amount of what the project/DL has received so far
![image](https://github.com/user-attachments/assets/575e499d-9e2f-494a-8d22-1534b14c495e)

**Comment by efstajas** [comment]
Result types:
- Projects
  - Claimed (matches by name, account ID)
  - Unclaimed (as you type anything with pattern of `**/**` or `github.com/**`, one of the results is `github.com/<search term>`
- Drip List (matches by name, account ID)
- ens name (as you type anything, one of the results is `< search term >.eth`
- Streams (matches by name)

**Comment by efstajas** [comment]
We need to drop the USD amounts because that data is not indexable in any way

**Comment by efstajas** [comment]
Show amount of supporters or even supporter pile

**Comment by efstajas** [comment]
Need design for loading

**Comment by brandonhaslegs** [comment]
@efstajas Is there a way to NOT show a ENS name result if we KNOW it's not a claimed ENS name? For example `iuflhdksjrusdjgsafkhga.eth` is not a ENS name and it would lead to a "not found" page, so would be nice to not even show it in results.

**Comment by efstajas** [comment]
Yeah should be possible, but it might slightly adversely affect the performance. But should be possible. We'd just try to resolve whatever is in the search bar as an ens name once the user stops typing for a short while, in parallel with the request to our own search engine

**Comment by brandonhaslegs** [comment]
so what we'll do is reserve this first spot for ENS result, and so it will always be loading until we find a result and if there's no result, then we can just keep the spot there and say no ENS result. but btw I don't think we should block showing other results while searching for the ENS result.

**Comment by brandonhaslegs** [comment]
All possible search results I think

![Image](https://github.com/user-attachments/assets/4443a2df-1c23-4a6e-aba0-839a75bd8862)
